### PR TITLE
arm64: dts: qcom: msm8916-bq-paella: Fix touchscreen

### PR DESCRIPTION
**Pull request by Minecrell**
Whenever a touch event is available, the touchscreen changes the GPIO
for the IRQ from high -> low (a falling edge). The GPIO is changed back
from low -> high once the touch event has been read with
HX852_READ_ALL_EVENTS.

At least on the BQ models, the touchscreen seems to send an empty
touch event after initial power up, so the GPIO switches to "low".
This is before the IRQ is enabled, so hx852_interrupt() is not triggered.

This means that we don't read the event from the controller
(HX852_READ_ALL_EVENTS) and therefore the GPIO does not change back
from low -> high. This is problematic because hx852_interrupt()
is only triggered when the GPIO switches from high -> low.
It is not triggered when the IRQ stays on low forever.

In other words, we will never get any further touch events from the
controller. :(

This can be avoided by making the IRQ level triggered (as on downstream).
Now, hx852_interrupt() is repeatedly called as long as the IRQ stays
low. This means that we read the pending touch event as soon as the IRQ
is enabled and then receive all further IRQs as well. Yay!

Cc: @JonnyMe

### CODE REVIEW

**Comment by JonnyMe** [comment]
Great!! Thank you @Minecrell! I will test this tomorrow, I wonder why it worked before

**Comment by Minecrell** [comment]
> I wonder why it worked before

I'm not sure either. Perhaps something is different on the Cyanogen Edition.

**Comment by JonnyMe** [comment]
Well, I just tested it and touch is working properly on the Cyanogen Edition :-) thank you!
